### ISSUE TITLE
provide wrappers for native functions supported by Cuda

### ISSUE THREAD

**Issue by XzzX** [kokkos/kokkos#5638]
Cuda provides more device optimized math functions (https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#mathematical-functions-appendix) than currently supported by Kokkos (#4767). In principle they can be provided if available and otherwise a fallback will be used.

**Comment by XzzX** [comment]
I can provide a MR, however, I have some questions regarding the implementation.  @dalg24 

Currently, a namespace is set depending on the enabled backend and functions from this namespace are used? Does this work if I enable some device backend and afterwards execute on the host? Or do I end up with device functions on host?  

Do you want it to be implemented similar to `abs` by providing overloads for a certain set of argument types?

**Comment by dalg24** [comment]
You probably will need to use the `KOKKOS_IF_ON_{HOST,DEVICE}` macros.

**Comment by crtrott** [comment]
Core is not there fft support belongs so no - fft doesn't go heee.

**Comment by XzzX** [comment]
The unit tests check against a reference function in `std` namespace. What do you want to do if the function is not available in `std` namespace? Inject a dummy function into `std` namespace? Since the unit tests are defined in a header file, I have some headaches with this.

**Comment by dalg24** [comment]
No you will need to have a different logic for these functions
```C++
KOKKOS_INLINE_FUNCTION double rcbrt(double x) {
#if defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_ENABLE_CUDA)
  KOKKOS_IF_ON_DEVICE( return ::rcbrt(x); )
  KOKKOS_IF_ON_HOST( ... )
#else
  ...
#endif
}

**Comment by dalg24** [comment]
Do not define any function in the `std::` namespace.  Yes we need custom logic for these new functions.

**Comment by ajpowelsnl** [comment]
@XzzX -- Have we addressed your needs with the merging of #5644?
